edgelet/hosting/src/apis/configuration.rs

```diff
@@ -0,0 +1,37 @@
+/*
+ * IoT Edge Hosting Environment API
+ *
+ * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
+ *
+ * OpenAPI spec version: 2019-04-10
+ *
+ * Generated by: https://github.com/swagger-api/swagger-codegen.git
+ */
+
+use failure::{format_err, Error};
+use hyper::client::connect::Connect;
+use hyper::{Client, Uri};
+
+pub struct Configuration<C: Connect> {
+    pub base_path: String,
+    pub user_agent: Option<String>,
+    pub client: Client<C>,
+    pub uri_composer: Box<dyn Fn(&str, &str) -> Result<Uri, Error> + Send + Sync>,
+    pub sas_token: Option<String>,
+}
+
+impl<C: Connect> Configuration<C> {
+    pub fn new(client: Client<C>) -> Configuration<C> {
+        Configuration {
+            base_path: "http://localhost".to_owned(),
+            user_agent: Some(format!("iotedge/{}", env!("CARGO_PKG_VERSION"))),
```

Comment:
If you do want an identifying user agent, use `edgelet_core::version_with_source_version()` instead of `env!("CARGO_PKG_VERSION")` since the latter will always be `"0.1.0"`

Reply:
All of this code was generated by swagger codegen. If you look at the workload and management crates, they have similar values for user_agent. I'll change it to your recommendation though.

Reply:
Yeah, I know those are busted too. We need to fix them one of these days.